File: app/src/main/java/de/rafaelmiranda/memory/model/Game.java
package de.rafaelmiranda.memory.model;

import de.rafaelmiranda.memory.themes.Theme;

/**
 * This is instance of active playing game
 * 
 * @author sromku
 */
public class Game {

	/**
	 * The board configuration
	 */
	public BoardConfiguration boardConfiguration;

	/**
	 * The board arrangment
	 */
	public BoardArrangement boardArrangement;

	/**
	 * The selected theme
	 */
	public Theme theme;

	public GameState gameState;

}

File: app/src/main/java/de/rafaelmiranda/memory/model/Game.kt
package de.rafaelmiranda.memory.model

import de.rafaelmiranda.memory.types.Assistants
import de.rafaelmiranda.memory.types.GameType

/**
 * This is instance of active playing game and all its quirks.
 */
class Game {

    /**
     * The board configuration
     */
    lateinit var boardConfiguration: BoardConfiguration

    /**
     * The board arrangment
     */
    lateinit var boardArrangement: BoardArrangement

    /**
     * The selected gameType
     */
    lateinit var gameType: GameType

    /**
     * All the help being used.
     */
    lateinit var assistants: Assistants
}
